### PR TITLE
New fields in protobuf definitions

### PR DESCRIPTION
**Pull request by hsorellana**
**What this PR does / Why we need it**:
This PR adds new fields and structures in the Protobuf definitions. Also improves the Makefile display when running the available commands.

**Which issue(s) this PR fixes**:
This PR is part of the development of [Backfill](https://github.com/googleforgames/open-match/issues/1240) feature request.

### CODE REVIEW

File: internal/app/backend/backend_service.go [hunk 1]
```diff
@@ -171,7 +170,7 @@ func synchronizeRecv(ctx context.Context, syncStream synchronizerStream, m *sync
 			}
 			stats.Record(ctx, totalBytesPerMatch.M(int64(proto.Size(match))))
 			stats.Record(ctx, ticketsPerMatch.M(int64(len(match.GetTickets()))))
-			err = stream.Send(&pb.FetchMatchesResponse{Match: match})
+			err := stream.Send(&pb.FetchMatchesResponse{Match: match})
```
Comment (akremsa) [inline internal/app/backend/backend_service.go:173]: why did you decide to create a new err variable here?

Comment (hsorellana) [reply internal/app/backend/backend_service.go:173]: that was an idea i had about moving all that if block out but wasn't completely sure about, and that colon just escaped me

**Review by Laremere — CHANGES_REQUESTED** [review]

File: internal/app/frontend/frontend_service.go [hunk 2]
```diff
@@ -180,3 +180,29 @@ func doWatchAssignments(ctx context.Context, id string, sender func(*pb.Assignme
 
 	return store.GetAssignments(ctx, id, callback)
 }
+
+// AcknowledgeBackfill is used to notify OpenMatch about GameServer connection info.
+// This triggers an assignment process.
+func (s *frontendService) AcknowledgeBackfill(ctx context.Context, req *pb.AcknowledgeBackfillRequest) (*pb.Backfill, error) {
+	return nil, nil
```
[the remark this answers is not in the document]
Comment (akremsa) [reply internal/app/frontend/frontend_service.go:187]: Agree, something like this _return nil, status.Errorf(codes.Unimplemented)_

File: internal/app/backend/backend_service.go [hunk 3]
```diff
@@ -70,25 +68,25 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 	}
 
 	// Error group for handling the synchronizer calls only.
-	eg, ctx := errgroup.WithContext(stream.Context())
+	errGroup, ctx := errgroup.WithContext(stream.Context())
```
[the remark this answers is not in the document]
Comment (hsorellana) [reply internal/app/backend/backend_service.go:71]: to have a more declarative variable name and/or usage and avoid to having to the var definition to understand it

File: internal/testing/fake_frontend.go [hunk 4]
```diff
@@ -54,3 +54,29 @@ func (s *FakeFrontend) GetTicket(ctx context.Context, req *pb.GetTicketRequest)
 func (s *FakeFrontend) WatchAssignments(req *pb.WatchAssignmentsRequest, stream pb.FrontendService_WatchAssignmentsServer) error {
 	return status.Error(codes.Unimplemented, "not implemented")
 }
+
+// AcknowledgeBackfill is used to notify OpenMatch about GameServer connection info.
+// This triggers an assignment process.
+func (s *FakeFrontend) AcknowledgeBackfill(ctx context.Context, req *pb.AcknowledgeBackfillRequest) (*pb.Backfill, error) {
+	return nil, nil
```
[the remark this answers is not in the document]
Comment (hsorellana) [reply internal/testing/fake_frontend.go]: Fixed!

File: internal/app/query/query_service.go [hunk 5]
```diff
@@ -134,6 +134,10 @@ func (s *queryService) QueryTicketIds(req *pb.QueryTicketIdsRequest, responseSer
 	return nil
 }
 
+func (s *queryService) QueryBackfills(req *pb.QueryBackfillsRequest, responseServer pb.QueryService_QueryBackfillsServer) error {
+	return nil
```
[the remark this answers is not in the document]
Comment (hsorellana) [reply internal/app/query/query_service.go]: Fixed!

File: internal/app/frontend/frontend_service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hsorellana) [reply internal/app/frontend/frontend_service.go]: Fixed!

File: api/frontend.proto [hunk 6]
```diff
@@ -81,6 +81,30 @@ message WatchAssignmentsResponse {
   Assignment assignment = 1;
 }
 
+message AcknowledgeBackfillRequest {
```
[the remark this answers is not in the document]
Comment (aLekSer) [reply api/frontend.proto:84]: Also please refer to this PR as an example:
https://github.com/googleforgames/open-match/pull/1273/files#diff-cb9fb1f207fd5b8c4ef3730293c848aa033832f8dd69ebb8f81d606af3cce387R208

File: internal/app/backend/backend_service.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (aLekSer) [reply internal/app/backend/backend_service.go:71]: Personally I like this presentation:
>The greater the distance between a name’s declaration and its uses, the longer the name should be.
https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_the_larger_the_identifiers_scope_the_larger_its_name
Thus as @hsorellana you can see declaration and usage would appear on the one screen.

File: internal/rpc/clients.go [hunk 7]
```diff
@@ -26,14 +26,12 @@ import (
 	"strings"
 	"time"
 
-	grpc_logrus "github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus"
-	"go.opencensus.io/plugin/ocgrpc"
-
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_logrus "github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus"
```
[the remark this answers is not in the document]
Comment (aLekSer) [reply internal/rpc/clients.go:30]: This should be moved to a separate PR.

File: api/query.proto [hunk 8]
```diff
@@ -98,4 +108,12 @@ service QueryService {
       body: "*"
     };
   }
+
+  // QueryBackfills gets a list of Backfill.
```
Comment (aLekSer) [inline api/query.proto:112]:
```suggestion
  // QueryBackfills gets a list of Backfills.
```

File: internal/app/backend/backend_service.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (hsorellana) [reply internal/app/backend/backend_service.go]: Fixed

File: Makefile [hunk 9]
```diff
@@ -15,37 +15,38 @@
 ## Open Match Make Help
 ## ====================
 ##
-## Create a GKE Cluster (requires gcloud installed and initialized, https://cloud.google.com/sdk/docs/quickstarts)
+## # Create a GKE Cluster (requires gcloud installed and initialized, https://cloud.google.com/sdk/docs/quickstarts)
```
[the remark this answers is not in the document]
Comment (hsorellana) [reply Makefile]: Fixed, sending in other PR

File: internal/rpc/clients.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (hsorellana) [reply internal/rpc/clients.go]: Resolved

File: api/messages.proto [hunk 10]
```diff
@@ -201,6 +201,40 @@ message Match {
   // Optional, depending on the requirements of the connected systems.
   map<string, google.protobuf.Any> extensions = 7;
 
+  // Backfill request which contains additional information to the match
+  // and contains an association to a GameServer
+  Backfill backfill = 8;
```
Comment (Laremere) [inline api/messages.proto:206]: beta feature warning

File: api/messages.proto [hunk 11]
```diff
@@ -201,6 +201,40 @@ message Match {
...
+
+  // AllocateGameServer signalise Director that Backfill is new and it should 
+  // allocate a GameServer, this Backfill would be assigned.
+  bool allocate_gameserver = 9;
```
Comment (Laremere) [inline api/messages.proto:210]: beta feature warning

File: api/query.proto [hunk 12]
```diff
@@ -75,6 +75,16 @@ message QueryTicketIdsResponse {
   repeated string ids = 1;
 }
 
+message QueryBackfillsRequest {
```
Comment (Laremere) [inline api/query.proto:78]: beta feature warning

Comment (Laremere) [reply api/query.proto:78]: Be consistent with request/response types:  Either they all get beta feature request warnings, or none of them do.  I don't have an opinion which way to go as long as they're consistent.

File: internal/app/backend/backend_service.go [hunk 13]
```diff
@@ -24,15 +24,13 @@ import (
 	"sync"
```
Comment (Laremere) [inline internal/app/backend/backend_service.go:24]: This file's changes are unrelated to this PR.

**Review by Laremere — APPROVED** [review]

**Review by Laremere — CHANGES_REQUESTED** [review]
hit wrong checkbox...

File: api/query.proto (hunk 12 above)
Comment (hsorellana) [reply api/query.proto:80]: Added beta feature warnings in all places but for some reason they are not showing the latest changes

**Review by Laremere — APPROVED** [review]

**Review by akremsa — APPROVED** [review]

**Review by aLekSer — APPROVED** [review]
Thanks Hugo,
Looks good to me.
